securing_apps/topics/oidc/java/servlet-filter-adapter.adoc

```diff
@@ -64,3 +64,91 @@ To use this filter, include this maven artifact in your WAR poms:
     <version>{project_versionMvn}</version>
 </dependency>
 ----
+ifeval::[{project_community}==true]
+===== Using on OSGi
+
+The servlet filter adapter is packaged as an OSGi bundle, and thus is usable in a generic OSGi environment with HTTP Service and HTTP Whiteboard.
+
+====== Installation
+
+The adapter and its dependencies are distributed as Maven artifacts, so you'll need either working Internet connection to access Maven Central, or have the artifacts cached in your local Maven repo.
+
+If you are using Apache Karaf, you can simply install a feature from the Keycloak feature repo:
+
+[source,subs="attributes+"]
+----
+karaf@root()> feature:repo-add mvn:org.keycloak/keycloak-osgi-features/{project_versionMvn}/xml/features
+karaf@root()> feature:install keycloak-servlet-filter-adapter
+----
+
+For other OSGi runtimes, please refer to the runtime documentation on how to install the adapter bundle and its dependencies.
+
+====== Configuration
+
+First, the adapter needs to be registered as an OSGi component. The easiest way to do that is to extend `org.keycloak.adapters.servlet.KeycloakOIDCFilter` and to decorate it with OSGi annotations:
+
+[source,java]
+----
+package mypackage;
+
+import javax.servlet.Filter;
+import org.keycloak.adapters.servlet.KeycloakOIDCFilter;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.http.whiteboard.HttpWhiteboardConstants;
+
+@Component(
+    immediate = true,
+    service = Filter.class,
+    property = {
+        KeycloakOIDCFilter.CONFIG_FILE_PARAM + "=" + "keycloak.json",
+        HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_PATTERN + "=" +"/*",
+        HttpWhiteboardConstants.HTTP_WHITEBOARD_CONTEXT_SELECT + "=" + "(osgi.http.whiteboard.context.name=mycontext)"
+    }
+)
+public class KeycloakFilter extends KeycloakOIDCFilter {
+  //
+}
+----
+
+To configure the component, either create a `mypackage.KeycloakFilter.cfg` file under the standard config location for your OSGi runtime:
```

Comment:
I would add: 

The above snippet uses OSGi declarative service specification to expose the filter as a declarative service component and as an OSGI service under `javax.servlet.Filter` class. 
Once the class is published in the OSGi service registry it is going to be picked up by OSGi HTTP Service implementation and used for filtering requests for the specified servlet context. This will trigger Keycloak adapter for every request that matches the servlet context path + the filter path. 

Since the component is using ConfigAdmin service you can configure it's properties dynamically like this: